pallets/market/src/deal_parameters.rs

```diff
@@ -0,0 +1,189 @@
+/// Adjusted to have the `None` options in [`OffchainDealDurationBound`]
+/// set to the chain enforced min and max
+#[derive(Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub struct DealDurationBound<BlockNumber> {
+    pub lower: BlockNumber,
+    pub upper: BlockNumber,
+}
+
+/// Deal Parameters submitted by the storage provider.
+/// This type gets converted to [`DealParameters`] to set chain
+/// defaults for the lower and upper deal bounds.
+#[derive(Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub struct OffchainDealParameters<Balance, BlockNumber> {
+    pub minimum_price_per_block: Balance,
+    pub deal_duration: OffchainDealDurationBound<BlockNumber>,
+}
+
+/// Deal Parameters submitted by the storage provider.
+/// After converting the [`OffchainDealParameters`] Options to chain enforced
+/// values.
+#[derive(Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub struct DealParameters<Balance, BlockNumber> {
+    pub minimum_price_per_block: Balance,
+    pub deal_duration: DealDurationBound<BlockNumber>,
+}
+
+impl<Balance, BlockNumber> DealParameters<Balance, BlockNumber>
+where
+    Balance: PartialOrd + frame_support::traits::tokens::Balance,
+    BlockNumber: sp_runtime::traits::BlockNumber,
+    // `Balance` and `BlockNumber` are not directly tied to their `Config` counterparts.
+    // The structure is flexible enough to be used for other purposes,
+    // so we limit the generics to the essential subset of traits only.
+{
+    /// Checks the deal parameters against the given deal parameters
+    /// Returns true if everything checks out
+    /// False if something is out of the duration bound
+    pub fn check_against_proposed_deal<Address>(
+        &self,
+        proposal: &DealProposal<Address, Balance, BlockNumber>,
+    ) -> bool {
+        let deal_duration = proposal.end_block - proposal.start_block;
+        proposal.storage_price_per_block >= self.minimum_price_per_block
+            && deal_duration >= self.deal_duration.lower
+            && deal_duration <= self.deal_duration.upper
+    }
+
+    /// Validates submitted deal parameters to be within the chains constants
+    pub fn validate(
+        &self,
+        chain_min_duration: BlockNumber,
+        chain_max_duration: BlockNumber,
+    ) -> bool {
+        Balance::zero() < self.minimum_price_per_block
+            && self.deal_duration.lower < self.deal_duration.upper
+            && (self.deal_duration.lower..=self.deal_duration.upper).contains(&chain_min_duration)
+            && (self.deal_duration.lower..=self.deal_duration.upper).contains(&chain_max_duration)
+    }
+}
+
+/// Converts [`OffchainDealParameters`] to [`DealParameters`]
+/// and fills in the `None` with the given chain constants.
+/// Conversion function instead of a `From` implementation because
+/// we need the chain constants in [`crate::Config`]
```

Comment:
This isn't a doc comment
```suggestion
// Conversion function instead of a `From` implementation because
// we need the chain constants in [`crate::Config`]
```